webapp/src/js/bootstrapper/index.js

```diff
@@ -224,53 +224,70 @@
       .catch(() => true);
 
     let isInitialReplicationNeeded;
-    Promise.all([swRegistration, testReplicationNeeded(), setReplicationId(POUCHDB_OPTIONS, localDb)])
-      .then(function(resolved) {
+    Promise
+      .all([swRegistration, testReplicationNeeded(), setReplicationId(POUCHDB_OPTIONS, localDb)])
+      .then(resolved => {
         purger.setOptions(POUCHDB_OPTIONS);
         isInitialReplicationNeeded = !!resolved[1];
 
         if (isInitialReplicationNeeded) {
+          window.startupTimes.replicationStarted = performance.now();
           return docCountPoll(localDb)
             .then(() => initialReplication(localDb, remoteDb))
             .then(testReplicationNeeded)
             .then(isReplicationStillNeeded => {
               if (isReplicationStillNeeded) {
                 throw new Error('Initial replication failed');
               }
-            });
+            })
+            .then(() => window.startupTimes.replicationEnded = performance.now());
         }
       })
       .then(() => {
         return purger
           .shouldPurge(localDb, userCtx)
           .then(shouldPurge => {
+            window.startupTimes.purgedDB = shouldPurge;
+
             if (!shouldPurge) {
               return;
             }
 
+            window.startupTimes.purgeStarted = performance.now();
             return purger
               .purge(localDb, userCtx)
               .on('start', () => setUiStatus('PURGE_INIT'))
               .on('progress', progress => setUiStatus('PURGE_INFO', { count: progress.purged }))
-              .catch(err => console.error('Error attempting to purge', err));
+              .catch(err => {
+                console.error('Error attempting to purge', err);
+                window.startupTimes.purgedDB = false;
```

Comment:
I think we could have a different value here, so `purgedDB` (or what it will be called) isn't binary. So we can add a different entry when we tried to purge but failed.